odo/backends/csv.py

```diff
@@ -39,7 +43,36 @@ def alias(key):
     return aliases.get(key, key)
 
 
+@contextmanager
+def open_file(path, *args, **kwargs):
+    f = compressed_open.get(ext(path), open)(path, *args, **kwargs)
+    try:
+        yield f
+    finally:
+        f.close()
+
+
+def infer_header(path, nbytes=10000, encoding='utf-8', **kwargs):
+    with open_file(path, 'rb') as f:
+        raw = f.read(nbytes)
+    return csv.Sniffer().has_header(raw if PY26 else raw.decode(encoding))
```

Comment:
I was burned sufficiently often by windows and py2/3 differences to avoid the csv module alltogether in favor of pandas.read_csv.  If you've handled that then that sounds good.

Curious, what triggered the move away here?